pkg/controller/kabaneroplatform/landing.go

```diff
@@ -70,27 +71,79 @@ func deployLandingPage(k *kabanerov1alpha1.Kabanero, c client.Client) error {
 		return err
 	}
 
-	// Create a clientset to drive API operations on resources.
-	config, err := clientcmd.BuildConfigFromFlags("", "")
+	// Retrieve the kabanero landing URL.
+	landingURL, err := getLandingURL(k, c)
 	if err != nil {
 		return err
 	}
 
-	clientset, err := kubernetes.NewForConfig(config)
+	// Create a Deployment. The landing application requires knowledge of the landingURL
+	// post route creation.
+	f, err = rev.OpenOrchestration("kabanero-landing-deployment.yaml")
 	if err != nil {
 		return err
 	}
 
-	// Retrieve the kabanero landing URL.
-	landingURL, err := getLandingURL(k, c, config)
+	s, err = renderOrchestration(f, templateContext)
 	if err != nil {
 		return err
 	}
 
-	// Create a Deployment. The landing application requires knowledge of the landingURL
-	// post route creation.
-	env := []corev1.EnvVar{{Name: "LANDING_URL", Value: landingURL}}
-	err = createDeployment(k, clientset, c, "kabanero-landing", image, env, nil, nil, kllog)
+	m, err = mf.FromReader(strings.NewReader(s), c)
+	if err != nil {
+		return err
+	}
+
+	transforms = []mf.Transformer{
+		mf.InjectOwner(k),
+		mf.InjectNamespace(k.GetNamespace()),
+		kabTransforms.AddEnvVariable("LANDING_URL", landingURL),
+	}
+
+	// See if we should define the OAuth volume and variables
+	secretInstance := &corev1.Secret{}
+	secretName := "kabanero-github-oauth-secret"
+	err = c.Get(context.Background(), types.NamespacedName{
+		Name:      secretName,
+		Namespace: k.ObjectMeta.Namespace}, secretInstance)
+
+	if err == nil {
+		// The secret exists, go ahead and put in the oauth information
+		transforms = append(transforms, kabTransforms.MountSecret(secretName, "/etc/oauth"))
```

Comment:
so for the scenario when a user creates the secret named `kabanero-github-oauth-secret` and adds two key value pairs (client ID, and secret). Do we need to tell them to name the keys x and y something specific?

The way I assume the secret is mounted to `/etc/oauth` is - it will create two files in the container at `/etc/oauth`, and the filenames are named after the keys given when creating the secret. I'll have to know those filenames for my automation to grab the right values out.

This is more a doc question if we will have to document to tell them name the keys `githubClientID` and `githubClientSecret` or something.

Reply:
Yes, I assumed that there would be doc that said something like "Create a secret named `kabanero-github-oauth-secret` and add two keys/values.  You will need to pick the keys and then the console will need to look for those in `/etc/oauth`.